lib/init/features/firestore.js

```diff
@@ -27,11 +34,62 @@ var _initRules = function(setup, config) {
       type: "input",
       name: "rules",
       message: "What file should be used for Firestore Rules?",
-      default: "firestore.rules",
+      default: DEFAULT_RULES_FILE,
     },
-  ]).then(function() {
-    return config.writeProjectFile(setup.config.firestore.rules, RULES_TEMPLATE);
-  });
+  ])
+    .then(function() {
+      var filename = setup.config.firestore.rules;
+
+      if (fsutils.fileExistsSync(filename)) {
+        var msg =
+          "File " +
+          chalk.bold(filename) +
+          " already exists." +
+          " Do you want to overwrite it with the Firestore Rules from the Firebase Console?";
+        return prompt.once({
+          type: "confirm",
+          message: msg,
+          default: false,
+        });
+      }
+
+      return Promise.resolve(true);
+    })
+    .then(function(overwrite) {
+      if (!overwrite) {
+        return Promise.resolve();
+      }
+
+      var projectId = setup.instance || setup.rcfile.projects.default;
+      return _getRulesFromConsole(projectId).then(function(contents) {
+        return config.writeProjectFile(setup.config.firestore.rules, contents);
+      });
+    });
+};
+
+var _getRulesFromConsole = function(projectId) {
+  return gcp.rules
+    .getLatestRulesetName(projectId)
+    .then(function(name) {
+      if (!name) {
+        logger.debug("No rulesets found, using default.");
+        var defaultContent = [{ name: DEFAULT_RULES_FILE, content: RULES_TEMPLATE }];
+        return Promise.resolve(defaultContent);
+      }
+
+      logger.debug("Found ruleset: " + name);
+      return gcp.rules.getRulesetContent(name);
+    })
+    .then(function(rules) {
+      for (var i = 0; i < rules.length; i++) {
+        var entry = rules[i];
+        if (entry.name == DEFAULT_RULES_FILE) {
```

Comment:
use `===`

Reply:
Not sure I understand what's going on here, why do you need to check that entry.name is the DEFAULT_RULES_FILE? If the rules are from the Console, then will this still be true?

Reply:
Technically the API supports multiple files per ruleset, even though right now we only support one.  So the result is an array of things with `{ name, content }` structure.  In case the user has somehow messed with things and gotten multiple files in there, the one we want is called `firestore.rules`.